x/ibc/02-client/keeper/client.go

```diff
@@ -28,13 +29,11 @@ func (k Keeper) CreateClient(
 		panic(fmt.Sprintf("client type is already defined for client %s", clientID))
 	}
 
-	clientState, err := k.initialize(ctx, clientID, clientType, consensusState, trustingPeriod, unbondingPeriod)
-	if err != nil {
-		return nil, sdkerrors.Wrapf(err, "cannot create client with ID %s", clientID)
-	}
+	height := consensusState.GetHeight()
+	k.SetClientConsensusState(ctx, clientID, height, consensusState)
```

Comment:
The spec still has an `initialise` function that can be defined per-client-type. Is there a reason we're altering this?

Example `initialise` function for ICS 7 [here](https://github.com/cosmos/ics/tree/master/spec/ics-007-tendermint-client#client-initialisation) - it still returns a `ClientState`. It's true that the `ConsensusState` should also be set - should we instead update the `initialise` function in the spec to be stateful?

Reply:
The Initialize function still exists in 07-tendermint. It just gets called in the handler, and then the initialized clientState gets passed into the clientKeeper to get persisted correctly. The `Initialize` function has to be stateful because it needs to initialize Tendermint ClientState with trustingPeriod and ubdPeriod.

In order to initialize the TM ClientState correctly, there needs to be a type switch on the msg and then a method that can take the TM-specific msg fields and construct the TM ClientState correctly

From an ergonomics point of view, it makes way more sense to me that we type switch on the msg right in the Handler where it is getting passed in. Rather than filtering the msg further down the call stack into the Keeper.CreateClient function (i.e. I don't think this keeper function should take in MsgCreateClient as an argument and type switch here)

Furthermore, I think the 07-tendermint package should be responsible for taking the concrete 07-tendermint `CreateClientMsg` and constructing the concrete `07-tendermint` ClientState. It seems weird that there would be a type switch of the msg in the ClientKeeper on and then a helper function in 02-client that will construct the 07-tendermint ClientState

IMO, the responsibility of creating a specific clientState from a specific CreateClientMsg should be delegated to the module that created those types. So if there's a new `08-poa` module that has its own ClientState and its own CreateClientMsg. The `02-Client` handler will simply add this type to its switch clause and then call an InitializeFromMsg function in `08-poa` which will do all the work of initializing an 08-poa specific clientstate. The client handler will then take this correctly initialized clientState and pass it to the Client Keeper which will persist all the data in the store without knowing the underlying specifics of the clientState

Reply:
This can be done in a later PR, let's just make an issue about it so we don't forget.